pkg/controller/operator/common/util.go

```diff
@@ -179,6 +179,31 @@ func VolumeRevisionLabelsModifierFactory(ctx context.Context, client ctrlruntime
 	}
 }
 
+// VersionLabelModifierFactory adds the version label for Deployments and their corresponding pods.
+func VersionLabelModifierFactory(version string) reconciling.ObjectModifier {
+	return func(create reconciling.ObjectReconciler) reconciling.ObjectReconciler {
+		return func(existing ctrlruntimeclient.Object) (ctrlruntimeclient.Object, error) {
+			obj, err := create(existing)
+			if err != nil {
+				return obj, err
+			}
+
+			deployment, ok := obj.(*appsv1.Deployment)
+			if !ok {
+				return obj, fmt.Errorf("VersionLabelModifier can only be used with deployments")
```

Comment:
`fmt.Errorf("VersionLabelModifier can only be used with deployments, not %T", obj)`

Reply:
(let's not make it sound like there is a technical reason version labels are not allowed/supported on other objects)

Reply:
sounds good. Fixed